### PR TITLE
vmalert: support multi-doc yaml parsing

### PR DESCRIPTION
**Pull request by jiekun**
### Describe Your Changes

Add support for https://github.com/VictoriaMetrics/VictoriaMetrics/issues/6753

### Checklist

The following checks are **mandatory**:

- [x] My change adheres [VictoriaMetrics contributing guidelines](https://docs.victoriametrics.com/contributing/).

### CODE REVIEW

File: app/vmalert/config/testdata/rules/rules-multi-doc-good.rules [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      - alert: VMRows
+        for: 1ms
+        expr: vm_rows > 0
+        labels:
+          label: bar
+          host: "{{ $labels.instance }}"
+        annotations:
+          summary: "{{ $value }}"
+---
+groups:
```
Comment (f41gh7) [inline app/vmalert/config/testdata/rules/rules-multi-doc-good.rules:15]: Could you also add a test for annotation with possible `---`:
```
        annotations:
          summary: \n markdown result is : ----\n # header\n body: \n text \n----\n"
```

Comment (jiekun) [reply app/vmalert/config/testdata/rules/rules-multi-doc-good.rules:15]: Good catch & I can get your point. Just updated the test cases, it seems the code still working.

File: app/vmalert/config/config_test.go [hunk 2]
```diff
@@ -58,7 +58,7 @@ groups:
 }
 
 func TestParse_Success(t *testing.T) {
-	_, err := Parse([]string{"testdata/rules/*good.rules", "testdata/dir/*good.*"}, notifier.ValidateTemplates, true)
+	_, err := Parse([]string{"testdata/rules/*multi-doc-good.rules", "testdata/dir/*good.*"}, notifier.ValidateTemplates, true)
```
Comment (Haleygo) [inline app/vmalert/config/config_test.go:61]: Looks like a temp change, could you revert this?

**Review by Haleygo — COMMENTED** [review]
Test is failing now, https://github.com/VictoriaMetrics/VictoriaMetrics/actions/runs/10776179069/job/29882389360?pr=6959
>--- FAIL: TestParse_Success (0.00s)
    config_test.go:63: error parsing files failed to parse [testdata/rules/*multi-doc-good.rules testdata/dir/*good.*]: errors(1): failed to parse file "testdata/rules/rules-multi-doc-good.rules": yaml: line 24: mapping values are not allowed in this context

File: app/vmalert/config/config_test.go (hunk 2 above)
Comment (jiekun) [reply app/vmalert/config/config_test.go]: Fixed. Thanks.

**Pull request closed without merging**